### ISSUE TITLE
[Bug] Doesn't work in NextJS

### ISSUE THREAD

**Issue by SaadBazaz** [jonathandion/react-tilt#26]
I get the following error on fresh install and importing it in a NextJS page:
```SyntaxError: Cannot use import statement outside a module```

My import code:
```
import { Tilt } from 'react-tilt';
```

Environment:
"next": "13.1.1",
"react": "18.2.0",
"react-dom": "18.2.0",

**Comment by Justkanye** [comment]
I got the same error as well. Have you been able to get a solution yet?

**Comment by SaadBazaz** [comment]
> I got the same error as well. Have you been able to get a solution yet?

I started using the react-parallax-tilt library instead.

**Comment by vicarb** [comment]
im having the same issue

**Comment by atenadadkhah** [comment]
Same issue

**Comment by karlhorky** [comment]
Yeah it looks like `tsup` / TypeScript has been incorrectly configured in #23 - not only a problem with Next.js - it indicates a problem generally for the types, including a conflict between:

1. the type of package being published: CommonJS - no `"type": "module"` in `package.json` or anything else to make this package ESM
2. the type of [the source files](https://unpkg.com/browse/react-tilt@1.0.2/dist/index.js): ESM, because you can see `import` keyword there

cc @jonathandion 

Check out [the Are The Types Wrong? page for `react-tilt`](https://arethetypeswrong.github.io/?p=react-tilt%401.0.2)

Error message:

```
The implementation resolved at the package uses ESM syntax, but the detected module kind is CJS. This will be an error in Node (and potentially other runtimes and bundlers). The module kind was decided based on the nearest package.json’s lack of a "type": "module" field.
```

<img width="791" alt="Screenshot 2023-05-05 at 11 11 37" src="https://user-images.githubusercontent.com/1935696/236419924-72c568ec-15d2-4c2a-985d-e370b96f64ea.png">

**Comment by karlhorky** [comment]
@jonathandion since it looks like you want to publish CommonJS instead of ESM (unless you're ready to switch your package to ESM), changing [the `target` in your `tsconfig.json`](https://github.com/jonathandion/react-tilt/blob/19435ec8192b7c31d545674344ca3b6ea21c3e65/packages/react-tilt/tsconfig.json#LL4C13-L4C13) to `"CommonJS"` would be one option.

Opened a PR here:

- https://github.com/jonathandion/react-tilt/pull/28

**Comment by ProchaLu** [comment]
I was able to successfully implement this using the `next@13.4.1-canary.1` version and the App Router in the `app/` directory. In case you'd like to see a small reproduction project, I've created one on [CodeSandbox](https://codesandbox.io/p/sandbox/react-tilt-3s5jyw?file=%2Fapp%2Fpage.tsx&selection=%5B%7B%22endColumn%22%3A32%2C%22endLineNumber%22%3A15%2C%22startColumn%22%3A32%2C%22startLineNumber%22%3A15%7D%5D)

![Screenshot 2023-05-05 at 11 56 31](https://user-images.githubusercontent.com/80746311/236428556-fa8946be-fca8-43d0-85af-ae590660875e.png)

**Comment by ProchaLu** [comment]
I encountered the error `SyntaxError: Cannot use import statement outside a module` while using the `pages/` directory in Next.js. I've created a small reproduction project on [CodeSandbox](https://codesandbox.io/p/sandbox/tilt-test-pages-r3mvnn?file=%2Fpages%2Findex.tsx&selection=%5B%7B%22endColumn%22%3A14%2C%22endLineNumber%22%3A24%2C%22startColumn%22%3A14%2C%22startLineNumber%22%3A24%7D%5D) to showcase the issue
![Screenshot 2023-05-05 at 12 08 55](https://user-images.githubusercontent.com/80746311/236431958-7938b2ba-efc4-47b5-abca-3cd694d31d40.png)

**Comment by miketeix** [comment]
I added ``` transpilePackages: ["react-tilt"]``` to my `next.config.js` to get this to work.
